core/runtime/src/main/java/io/quarkiverse/langchain4j/guardrails/OutputGuardrail.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkiverse.langchain4j.guardrails;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.rag.AugmentationResult;
+
+public interface OutputGuardrail {
+
+    class ValidationException extends Exception {
+
+        private final boolean retry;
+        private final String reprompt;
+
+        public ValidationException(String message, boolean retry, String reprompt) {
+            super(message);
+            this.retry = retry;
+            this.reprompt = reprompt;
+        }
+
+        public boolean isRetry() {
+            return retry;
+        }
+
+        public String getReprompt() {
+            return reprompt;
+        }
+    }
+
+
+    default void validate(AiMessage responseFromLLM) throws ValidationException {
+        validate(responseFromLLM, null, null);
+    }
+
+    void validate(AiMessage responseFromLLM, ChatMemory memory, AugmentationResult augmentationResult) throws ValidationException;
```

Comment:
I would much prefer for API evolution reasons to have all these be part of a record or class instead of multiple params

Reply:
Will do that tomorrow while writing the documentation.

Reply:
Actually, just did it.